tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -1121,3 +1124,133 @@ def test_workflows_halts_on_multiple_exact_matches(workflow_app):
 
     assert obj.status == ObjectStatus.HALTED
     assert obj.extra_data['_action'] == 'resolve_multiple_exact_matches'
+
+
+def fake_validation(data, schema=None):
+    return
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+@mock.patch('inspirehep.modules.workflows.tasks.matching.match',
+            return_value=iter([]))
+@mock.patch('inspirehep.modules.workflows.tasks.actions.validate',
+            side_effect=[fake_validation, validate])
+def test_workflow_with_validation_error(
+    fake_validation,
+    mocked_match,
+    mocked_magpie_json_api_request,
+    mocked_beard_json_api_request,
+    workflow_app,
+):
+    RECORD_WITH_VALIDATION_ERROR = {
+        '$schema': 'https://labs.inspirehep.net/schemas/records/hep.json',
+        'titles': [
+            {
+                'title': 'Update without conflicts title.'
+            },
+        ],
+        'arxiv_eprints': [
+            {
+                'categories': [
+                    'WRONG_CATEGORY',
+                    'hep-th'
+                ],
+                'value': '1703.04802'
+            }
+        ],
+        'document_type': ['article'],
+        '_collections': ['Literature'],
+        'acquisition_source': {'source': 'arXiv'},
+    }
+    workflow = build_workflow(RECORD_WITH_VALIDATION_ERROR)
+    with requests_mock.Mocker() as requests_mocker:
+        batch_uploader = re.compile('http://localhost:1234/batchuploader/robotupload/*')
+        holidingpen = re.compile('http://test-indexer:9200/holdingpen-hep/hep/*')
+        records = re.compile("http://test-indexer:9200/records-hep/hep/*")
+        ticket = re.compile("http://rt.inspire/ticket/new*")
```

Comment:
do we really need to mock all these requests?

Reply:
Yes, we need response 200 from them. I started workflow and I was catching all requests which appeared and was making problems.